### DISCUSSION TITLE
Syntax highlighting for relative code files (Not Embedded ones)

### DISCUSSION THREAD

**Discussion by ghost** [community/community#23810] [Other Feature Feedback, Questions, & Ideas]
<p>Hi, I was just wondering whether there is a workaround to enable syntax highlighting for relative links that point to a code file in the repository.</p>
<p>For example, <code>[foo](bar.cpp)</code> should enable C++ syntax highlighting for <code>bar</code>.</p>
<p>I searched around and couldn’t really find anything. Just wanted to confirm with the community.</p>
<p>Thanks!</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>Hello <a class="mention" href="/just-a-visitor">@Just-A-Visitor</a>, thanks for joining us.</p>
<p>If you’re talking about displaying code via GitHub Pages, you’re correct, there isn’t any built-in syntax highlighting support for full files. The way you linked to the source code file there, you’re essentially telling the browser to download it as a raw text file. What you could do instead is link to the source code file on GitHub itself via an absolute URL or there may be an extension for <a href="https://jekyllrb.com">Jekyll</a> (or whatever other static website generator you might be using) that would do this syntax highlighting work for you.</p>
<p>I hope that helps!</p>

**Comment by ghost** [comment]
<p>For anyone stumbling upon this topic in future, I found a <a href="https://chrome.google.com/webstore/detail/redirector/ocgpenflpmgnfapjedencafcfakcekcd/related?hl=en" rel="nofollow noopener">chrome-extension</a> to redirect code files automatically to Github.</p>
